src/Server.cpp

```diff
@@ -463,9 +449,9 @@ void Server::parseHeader(const std::string& requestString, HTTPRequest& request)
  * @param bodyString The body string to parse.
  * @param request The HTTPRequest object to store the parsed body.
  */
-void Server::parseBody(const std::string& bodyString, HTTPRequest& request)
+void Server::parseBody(const std::string& bodyString, HTTPRequest& request, std::vector<char>& buffer)
```

Comment:
Small thing: since we always use the server buffer we could directly pass it inside the dispatch function:
`m_requestParser.parseBody(bodyString, request, getBuffer());`